### ISSUE TITLE
AbstractFile#getURL does not correctly handle spaces when converting from URL to file

### ISSUE THREAD

**Issue by adriaanm** [scala/bug#10764]
Jason reports in https://github.com/scala/scala/pull/6314#issuecomment-371385148:

> I'm not 100% sure, but our nightly build on windows seems to have [started failing](https://scala-ci.typesafe.com/job/scala-2.12.x-integrate-windows/896/) after this change.

> Example error:
> ```
> !!  255 - run/macro-duplicate                       [compilation failed]
> ##### Log file 'Y:\jenkins\workspace\scala-2.12.x-integrate-windows\test\files\run\macro-duplicate-run.log' from failed test #####
> Test_2.scala:5: error: macro implementation not found: foo
> (the most common reason for that is that you cannot use macro implementations in the same compilation run that defines them)
>   Macros.foo
>          ^
> one error found
> ```

**Comment by lrytz** [comment]
The issue is entries in the classpath with spaces. It's not windows-specific, the following is on my mac:

```
 my dir git:(2.12.x) ✗ pwd
/Users/luc/scala/scala/sandbox/my dir
 my dir git:(2.12.x) ✗ cat A.scala
import scala.reflect.macros.blackbox.Context
import scala.language.experimental.macros

object M {
  def impl(c: Context): c.Expr[Int] = {
    import c.universe._
    println("makroing")
    c.Expr(q"1")
  }
  def foo: Int = macro impl
}
 my dir git:(2.12.x) ✗ cat B.scala
object Test extends App {
  println(M.foo)
}
 my dir git:(2.12.x) ✗ scalac A.scala && scalac B.scala && scala Test
makroing
1
 my dir git:(2.12.x) ✗ qsc A.scala && qsc B.scala && scala Test
B.scala:2: error: macro implementation not found: foo
(the most common reason for that is that you cannot use macro implementations in the same compilation run that defines them)
  println(M.foo)
            ^
one error found
```

https://github.com/scala/scala/blob/2ddc01b416390baccf51c95ad98920e4607203c1/src/compiler/scala/tools/nsc/typechecker/Macros.scala#L66-L74

`global.classPath.asURLs` contains `file:/Users/luc/scala/scala/sandbox/my%20dir/./`. Passing this to `AbstractFile.getURL` returns `null`, causing an NPE (which is caught in MacroRuntimes).

@jvican do you have time to come up with a fix?

**Comment by jvican** [comment]
Yes, looking into it. Will be ready by tomorrow morning, most likely.

**Comment by jvican** [comment]
The fix is up in #6400. Thanks for the detailed investigation @lrytz.

**Comment by adriaanm** [comment]
Fixed in https://github.com/scala/scala/pull/6400
